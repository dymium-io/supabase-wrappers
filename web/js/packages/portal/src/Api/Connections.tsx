import Alert from 'react-bootstrap/Alert'
import * as types from '@dymium/common/Types/Common'
import * as internal from '@dymium/common/Types/Internal'
import * as com from '../Common'

export function getConnections(setSpinner, setConns, setAlert, remap:internal.ConnectionMap|undefined, onSuccess) {
    setSpinner(true)
    setConns([])
    com.sendToServer("GET", "/api/getconnections",
        null, "",
        resp => {

            resp.json().then(_js => {
                let js = types.ConnectionResponse.fromJson(_js)
                if (js.status !== "OK") {
                    setAlert(
                        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
                            Error retrieving connections: {js.errormessage} { }
                        </Alert>
                    )
                    setTimeout(() => setSpinner(false), 500)
                    return
                }
            
                let cc = js.data.map(x => {
                    let ob = types.ConnectionRecord.fromJson({
                        id: x.id,
                        credid: x.credid,
                        dbtype: x.dbtype,
                        name: x.name,
                        dbname: x.dbname,
                        address: x.address,
                        port: x.port,
                        description: x.description,
                        useTLS: x.useTLS,

                    })
                    if(undefined != remap && x.name != null)
                        remap[x.name] = ob
                    return ob
                })

                setConns(cc)
                if(onSuccess != undefined) {
                    onSuccess()
                }
            })

            setTimeout(() => setSpinner(false), 500)
        },
        resp => {
            console.log("on error")
            setSpinner(false)
        },
        error => {
            console.log("on exception: " + error)
            setSpinner(false)
        })
}