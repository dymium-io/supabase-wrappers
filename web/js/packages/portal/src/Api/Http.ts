
export function sendToServer(method: string, url: string,
    headers, body: string, onsuccess, onfailure, onexception) {
    let token = window.sessionStorage.getItem("Session");

    if (token === null) {
        onfailure(null)
        return;
    }
    let _headers = {
        Authorization: "Bearer " + token,
        Cache: "no-cache",
    }
    if (headers != null) {
        _headers = {
            ..._headers,
            ...headers
        }
    }
    let params: any = {
        method,
        headers: _headers
    }
    if (body != "") {
        params = { ...params, body: body }
    }
  

    fetch(url, params
    ).then(response => {

        if (!response.ok) {
            onfailure(response)
        } else {
            onsuccess(response)
        }
    }).catch(onexception)
}
