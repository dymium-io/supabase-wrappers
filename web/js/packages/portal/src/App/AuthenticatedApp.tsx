import React, { useEffect } from 'react';
import { Outlet } from "react-router-dom";
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import Backdrop from "../Backdrop"
import Menu from './Menu'
import Auth from '../Auth'
import { Link } from "react-router-dom";
import Sidebar from './Sidebar'
import * as com from '../Common'

const AuthenticatedApp = () => {
  let roles = com.getTokenProperty("roles")
  if (roles === null) {
    return <div className="py-0 my-0">
      <Backdrop />
      <div className="text-center" style={{
        position: 'absolute', top: '0px', left: '0px',
        width: '100%', height: '100vh'
      }}>
        <div id="loginbox" >
          <div className="row">
            <div className="col-sm ml-5  text-center">
              <h1 style={{ marginTop: '1.3em', fontSize: '5em', fontWeight: '300' }} className="logoheader ">
                <Link className="linklink" to="/" > Dymium </Link></h1>
            </div>
            <div className="col-sm mt-5  text-center">
              <h1 className="mt-5 logoheader">Error</h1>
              <h1 className="unauthorized" >Not authorized!</h1>
              <div className="pt-2 logofooter">
                You are succesfully authenticated but not authorized to use Dymium. Please contact your administrator.
              </div>

            </div>
          </div>

        </div>
      </div>



    </div>
  }
  return (
    <>
      <Auth />
      <Menu />

      <Row style={{ height: '100vh' }} className="p-0 m-0">
        <Col xs="auto" id="sidebar_container" className="px-0 sidebar_container"><Sidebar /></Col>
        <Col xs className="p-0 spider" style={{ overflowX: "scroll" }}>
          <Outlet />
        </Col>

      </Row>

    </>
  )
}

export default AuthenticatedApp;