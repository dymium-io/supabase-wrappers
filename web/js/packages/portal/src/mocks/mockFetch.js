

export default async function mockFetch(url, js) {
    console.log("in mock fetch: ")
    switch (url) {
        case "/api/getconnections": {
            return  new Promise( (resolve, reject) => resolve(
                {
                    ok: true,
                    status: 200,
                    json:  () => new Promise( (resolve, reject) => resolve(JSON.parse('{"status":"OK","errormessage":"","data":[{"name":"adventureworks","dbtype":"PostgreSQL","address":"docker.for.mac.host.internal","port":5432,"dbname":"Adventureworks","useTLS":false,"description":"edited test data base from Microsoft","username":"postgres","password":null,"id":"a2a4fce3-1db5-47ea-a084-4601f880b9ba","credid":"2bd1e042-f5d1-4bcd-8563-7058f110c546"},{"name":"northwind","dbtype":"PostgreSQL","address":"docker.for.mac.host.internal","port":5432,"dbname":"northwind","useTLS":false,"description":"another MS database edited with password","username":"newusername","password":null,"id":"b157fc33-64ad-4173-a54b-b652ef21e0b7","credid":"876e712c-34ba-4bf0-a315-f3784b204512"}]}') ))
                }
              )
            )
        }
        case "/api/getdatascopes" : {
            return  new Promise( (resolve, reject) => resolve(
                {
                    ok: true,
                    status: 200,
                    json:  () => new Promise( (resolve, reject) => resolve(JSON.parse('{"status":"OK","errormessage":"","records":[{"name":"test1","id":"50fe96d1-b1a4-48df-87bd-6aced2ab1f5a"},{"name":"test2","id":"b3fddf89-5822-4227-9948-3ecb3c8c2e33"}]}') ))
                }
              )
            )            
        }
        case "/api/getmappings" : {
            return  new Promise( (resolve, reject) => resolve(
                {
                    ok: true,
                    status: 200,
                    json:  () => new Promise( (resolve, reject) => resolve(JSON.parse('{"status":"OK","errormessage":"","records":[{"id":"788b3c71-5371-41e8-b135-8037bcfcfd9f","dymiumgroup":"foo","directorygroup":"bar","comments":"Admin group fubar"},{"id":"27713f32-6772-4509-9476-41099b11cb18","dymiumgroup":"dada","directorygroup":"production","comments":"Comment changed via test"}]}') ))
                }
              )
            )                     
        }
        
        case "/api/getgroupsfordatascopes" : {
            return  new Promise( (resolve, reject) => resolve(
                {
                    ok: true,
                    status: 200,
                    json:  () => new Promise( (resolve, reject) => resolve(JSON.parse('[{"id":"50fe96d1-b1a4-48df-87bd-6aced2ab1f5a","name":"test1","groupid":"788b3c71-5371-41e8-b135-8037bcfcfd9f","groupname":"foo"}]') ))
                }
              )
            )                     
        }        
        default: {
            throw new Error(`Unhandled request: ${url}`);
        }
    }
}

