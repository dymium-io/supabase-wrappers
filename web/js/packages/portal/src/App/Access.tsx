import React, { useEffect, useState, useRef } from 'react';
import Tabs from 'react-bootstrap/Tabs'
import Tab from 'react-bootstrap/Tab'
import { useLocation, useNavigate } from "react-router-dom";
import Form from 'react-bootstrap/Form'
import Button from 'react-bootstrap/Button'
import Row from 'react-bootstrap/Row'
import Col from 'react-bootstrap/Col'
import { Link } from "react-router-dom";
import Alert from 'react-bootstrap/Alert'
import Spinner from '@dymium/common/Components/Spinner'
import * as http from '@dymium/common/Api/Http'
import * as com from '../Common'
import * as types from '@dymium/common/Types/Common'
import * as internal from '@dymium/common/Types/Internal'
import { useInitialize } from '../Utils/CustomHooks'
import { useAppDispatch, useAppSelector } from './hooks'
import { setActiveAccessTab } from '../Slices/menuSlice'

function YourDatascopes() {
  const [spinner, setSpinner] = useState(false)
  const [alert, setAlert] = useState<JSX.Element>(<></>)
  const [datascopes, setDatascopes] = useState<types.UserDatascopes>()
  let port = com.getTokenProperty("port")
  //"/api/getdatascopesaccess"
  let displayDatascope = (index) => {

  }
  let commandLine = (scope) => {
    let c = `psql -h localhost -p ${port} -d ${scope.name} -U ${datascopes !== undefined && datascopes.username}`

    return <div style={{ display: "flex" }}> <div className=" terminal">
      &gt;{c}
      </div>
      <i onClick={copy(c)} className="fas fa-copy clipbtn"></i>
    </div>
  }
  let python = scope => {
    let p = `conn = psycopg2.connect("host=localhost port=${port} dbname=${scope.name} user=${datascopes !== undefined && datascopes.username} password=${datascopes !== undefined && datascopes.password}")`

  return <div style={{ display: "flex" }}> <div className=" terminal">
        {p}
      </div>
      <i onClick={copy(p)} className="fas fa-copy clipbtn"></i>
    </div>
  }
let java = scope => {
  let j = `
  String url = "jdbc:postgresql://localhost:`+ port + `/`+scope.name+`";
  Properties props = new Properties();
  props.setProperty("user","${datascopes !== undefined && datascopes.username}");
  props.setProperty("password","${datascopes !== undefined && datascopes.password}");
  
  `

return <div style={{ display: "flex" }}> 
  <div><div className=" terminal"> <pre><code>{j}
  </code></pre></div>
  </div> 
  <i onClick={copy(j)} className="fas fa-copy clipbtn"></i>  
  </div>
}
let displayDatascopes = () => {
  let copy = () => {
    return e => {
      if (datascopes !== undefined)
        navigator.clipboard.writeText(datascopes.password);
    }
  }

  return <div>
    <div className="viewport">
      <div className="thickblue mb-2" style={{ fontWeight: 'bold' }}><i className="fa fa-key mr-2" aria-hidden="true"></i>Credentials:</div>
      <div>Username: {datascopes !== undefined && datascopes.username}</div>
      <div>Password: {datascopes !== undefined && datascopes.password} 
      <Button onClick={regenerate} style={{ marginTop: '-4px', paddingBottom: '0.1em', paddingTop: '0.0em' }} variant="dymium" className="mx-2" size="sm">Regenerate</Button> 

       {datascopes !== undefined && datascopes.password !== `**********` &&
        
     
        <i onClick={copy()}  className="fas fa-copy clipbtn"></i>

      } </div>
      <div className="mt-3" style={{ fontSize: '0.8em' }}>Password is only visible to you for a day. You can always generate a new one.</div>
    </div>
    {datascopes !== undefined && datascopes.datascopes.map(x => {
      return <div className="my-5">
        <h5 className="mb-2"><i className="fa fa-unlock mr-2" aria-hidden="true"></i>Ghost Database: {x.name}</h5>
        <div className="datascopeuse">
          <Tabs
            id="datascope"
            unmountOnExit={true} className=" mt-0 text-left ">
            <Tab eventKey="cmd" title="Command Line" className=" mx-2">
              {commandLine(x)}
            </Tab>
            <Tab eventKey="python" title="Python" className=" mx-2">
              {python(x)}
            </Tab>
            <Tab eventKey="java" title="Java" className=" mx-2">
              {java(x)}
            </Tab>

          </Tabs>
        </div>
      </div>
    })}
  </div>
}
let getDatascopeAccess = () => {
  let error = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
    Error retrieving datascopes.
  </Alert>
  setSpinner(true)
  http.sendToServer("GET", "/api/getdatascopesaccess",
    null, "",
    resp => {
      resp.json().then(js => {
        setDatascopes(types.UserDatascopes.fromJson(js))
        setSpinner(false)
      }).catch((error) => {
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  
        )
        setSpinner(false)
      })
    },
    resp => {
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>Get datascope access failed</Alert>                  

      )
      setSpinner(false)
    },
    error => {
      console.log("on exception")
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  

      )
      setSpinner(false)
    })
}

let regenerate = () => {
  let error = <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>
    Error regenerating password.
  </Alert>
  setSpinner(true)
  http.sendToServer("GET", "/api/regenpassword",
    null, "",
    resp => {
      resp.json().then(js => {
        setDatascopes(types.UserDatascopes.fromJson(js))
        setSpinner(false)
      }).catch((error) => {
        setAlert(
          <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>                  
        )
        setSpinner(false)
      })
    },
    resp => {
      setSpinner(false)
      setAlert(  <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>"Regenerate password failed"</Alert>        
      )
      setSpinner(false)
    },
    error => {
      console.log("on exception")
      setSpinner(false)
      setAlert(
        <Alert variant="danger" onClose={() => setAlert(<></>)} dismissible>{error.message}</Alert>        
      )
      setSpinner(false)
    })
}

useEffect(() => {
  getDatascopeAccess()
}, [])

return (
  <div className=" text-left">
    {alert}
    <h5 > Your Ghost Databases <Spinner show={spinner} style={{ width: '28px' }}></Spinner></h5>
    <div className=" text-left">

      {datascopes !== undefined && displayDatascopes()}


    </div>
  </div>
)
}
let orgid = com.getTokenProperty("schema")
let url = window.location.protocol + "//" + window.location.host + "/"
let params = `-c ${orgid} -p ${url}`

function copy(str) {

  return e => {
    navigator.clipboard.writeText(str);
  }
}
function Downloads() {

  return (
    <div className=" text-left">

      <h5 > Client Downloads</h5>
      <div className=" text-left">

        <div className="viewport">
          <div>On MS Windows:</div>
          <a href="/DymiumInstaller.exe" download="DymiumInstaller.exe"> <i className="fab fa-windows mr-2" aria-hidden="true"></i>Click to Download MS Windows Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;dymium.exe {params}
              </div>
              <i onClick={copy("dymium.exe " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>


        <div className="viewport">
          <div>On Mac OS X:</div>
          <a href="/DymiumInstaller.pkg" download="DymiumInstaller.pkg"> <i className="fab fa-apple mr-2" aria-hidden="true"></i>Click to Download Mac OS X Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;./dymium {params}
              </div>
              <i onClick={copy("./dymium " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>

        <div className="viewport">
          <div>On Linux:</div>
          <a href="/tunnel.tar.gz" download="tunnel.tar.gz"> <i className="fab fa-linux mr-2" aria-hidden="true"></i>Click to Download Linux Client</a>
          <div>
            CLI Usage:
            <div style={{ display: "flex" }}>
              <div className="terminal">
                &gt;./dymium {params}
              </div>
              <i onClick={copy("./dymium " + params)} className="fas fa-copy clipbtn"></i>
            </div>
          </div>
        </div>
        <div>
        To use the GUI, launch dymiumgui.app on Mac OS, or dymiumgui.exe on Windows.
        </div><div>
        <img src="/dymiumgui.png" style={{width: "540px"}}></img>  
        <div  className="mb-5">
        Paste the connection string for your OS into the application, and click Connect.
        </div>
        </div>
      </div>
    </div>
  )
}
function useQuery() {
  const { search } = useLocation();

  return React.useMemo(() => new URLSearchParams(search), [search]);
}
export default function Access() {
  const navigate = useNavigate();
  let t = useAppSelector((state) => {

    return state.reducer.activeAccessTab
  }
  )
  console.log("t=", t)
  const appDispatch = useAppDispatch()
  let query = useQuery();

  useEffect(() => {
    if (query.get("key") !== undefined) {

      appDispatch(setActiveAccessTab(query.get("key")))
      navigate("/app/access")
    }
  }, [])
  let tt = query.get("key")
  if (tt !== null) {
    t = tt
  }
  if (t == null) {
    t = "datascopes"
  }
  return (
    <Tabs defaultActiveKey={t}
      onSelect={(k) => appDispatch(setActiveAccessTab(k))}

      unmountOnExit={true} className="mb-3 text-left">
      <Tab eventKey="datascopes" title="Your Ghost Databases" className="mx-4">
        <YourDatascopes />
      </Tab>
      <Tab eventKey="download" title="Client Download And Use" className="mx-4">
        <Downloads />
      </Tab>

    </Tabs>
  )
}