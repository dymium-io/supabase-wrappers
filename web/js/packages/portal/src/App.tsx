import React, { useEffect } from 'react';
import LandingPage from './LandingPage'
import AuthenticatedApp from './App/AuthenticatedApp'
import Error404 from "./Error404"
import Logout from "./Logout"
import Error from "./Error"
import Invitation from "./Invitation"
import Sticky from "./App/Sticky"
import Dashboard from "./App/Dashboard"
import Customers from "./App/Customers"
import Connections from "./App/Connections"
import Datascopes from "./App/Datascopes"
import Groups from "./App/Groups"
import Rules from "./App/Rules"
import TestSQL from "./App/TestSQL"
import * as com from './Common'

import "@fontsource/roboto";
import 'bootstrap/dist/css/bootstrap.min.css';
import 'react-bootstrap-typeahead/css/Typeahead.css';

import { BrowserRouter, Routes, Route } from "react-router-dom";
//import Helmet from "react-helmet";

import '@dymium/common/App.scss';


function App() {
  return (
    <div className="App" id="app">

      <BrowserRouter>
        <Routes>

          <Route  path="/" element={<LandingPage />} />

          <Route path="/app/" element={<AuthenticatedApp />} >
            <Route path="/app/dashboard" element={<Dashboard />} />
            <Route path="/app/connections" element={<Connections />} />     
            <Route path="/app/datascopes" element={<Datascopes />} />                
            <Route path="/app/groups" element={<Groups />} />    
            <Route path="/app/rules" element={<Rules />} />         
            <Route path="/app/test" element={<TestSQL />} />                                             
            <Route path="/app/" element={<Sticky />} />          
          </Route>
          <Route  path="/app/logout" element={<Logout/>} />
          <Route path="/app/error" element={<Error />} />
          <Route path="/login" element={<Invitation />} />
          <Route path="*" element={<Error404 />} />
        </Routes>
      </BrowserRouter>

    </div>
  );
}

export default App;
