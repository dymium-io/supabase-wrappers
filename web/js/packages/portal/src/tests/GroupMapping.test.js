/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';

import { BrowserRouter } from "react-router-dom";
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux'
import store from '../Store/index'
import { Http } from '@dymium/common/Api/Http'
import  GroupMapping  from '../App/Groups'
import 'bootstrap/dist/css/bootstrap.min.css';
import '@dymium/common/App.scss';
import "@fortawesome/fontawesome-free/css/all.min.css";
import mockFetch from "../mocks/mockFetch";


beforeEach(() => {
  global.fetch = jest.fn().mockImplementation(() => mockFetch)();
  global.sessionStorage.setItem("Session", "mockJWT");
})

afterEach(() => {
   jest.restoreAllMocks()
});

test('test-sql', async () => {
    const component = await act( async () => renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                <GroupMapping />
            </BrowserRouter>
        </Provider >
    ));
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});