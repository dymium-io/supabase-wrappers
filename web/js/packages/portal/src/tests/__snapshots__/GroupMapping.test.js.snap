// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-sql 1`] = `
Array [
  <nav
    className="mb-3 text-left nav nav-tabs"
    onKeyDown={[Function]}
    role="tablist"
  >
    <a
      aria-selected={true}
      className="nav-item nav-link active"
      data-rb-event-key="groups"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="tab"
    >
      Group Mapping
    </a>
  </nav>,
  <div
    className="tab-content"
  >
    <div
      aria-hidden={false}
      aria-labelledby={null}
      className="fade mx-4 tab-pane active show"
      id={null}
      role="tabpanel"
    >
      <h5>
         Group mapping from your directory to Dymium
        <img
          className="hiddenspinner"
          src="/logo.png"
          style={
            Object {
              "width": "28px",
            }
          }
        />
      </h5>
      <div
        className="text-left mt-0 pt-0"
      >
        <div
          className="d-flex"
        >
          <div
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <label
              className="search-label"
              htmlFor="search-bar-0"
            >
              <span
                className="sr-only"
                id="search-bar-0-label"
              >
                Search this table
              </span>
              <input
                aria-labelledby="search-bar-0-label"
                className="form-control "
                id="search-bar-0"
                onChange={[Function]}
                onKeyUp={[Function]}
                placeholder="Search"
                style={Object {}}
                type="text"
                value=""
              />
            </label>
            <button
              className="btn btn-default "
              onClick={[Function]}
            >
              Clear
            </button>
          </div>
        </div>
        <div
          className="d-block"
        >
          <div
            className="react-bootstrap-table"
          >
            <table
              className="table table-striped table-sm"
              id="scaledtable"
            >
              <thead>
                <tr>
                  <th
                    aria-label="Directory group: sortable"
                    className="sortable"
                    onClick={[Function]}
                    onKeyUp={[Function]}
                    tabIndex={0}
                  >
                    Directory group:
                    <span
                      className="order-4"
                    />
                  </th>
                  <th
                    aria-label="Dymium group sortable"
                    className="sortable"
                    onClick={[Function]}
                    onKeyUp={[Function]}
                    tabIndex={0}
                  >
                    Dymium group
                    <span
                      className="order-4"
                    />
                  </th>
                  <th
                    aria-label="Description: sortable"
                    className="sortable"
                    onClick={[Function]}
                    onKeyUp={[Function]}
                    tabIndex={0}
                  >
                    Description:
                    <span
                      className="order-4"
                    />
                  </th>
                  <th
                    style={
                      Object {
                        "width": "50px",
                      }
                    }
                    tabIndex={0}
                  >
                    Edit
                  </th>
                  <th
                    style={
                      Object {
                        "width": "90px",
                      }
                    }
                    tabIndex={0}
                  >
                    Delete
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  className={null}
                  style={Object {}}
                >
                  <td>
                    bar
                  </td>
                  <td>
                    foo
                  </td>
                  <td>
                    Admin group fubar
                  </td>
                  <td
                    style={
                      Object {
                        "height": "30px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <i
                      className="fas fa-edit ablue"
                      onClick={[Function]}
                      role="button"
                    />
                  </td>
                  <td
                    style={
                      Object {
                        "height": "30px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <i
                      className="fas fa-trash ablue"
                      onClick={[Function]}
                      role="button"
                    />
                  </td>
                </tr>
                <tr
                  className={null}
                  style={Object {}}
                >
                  <td>
                    production
                  </td>
                  <td>
                    dada
                  </td>
                  <td>
                    Comment changed via test
                  </td>
                  <td
                    style={
                      Object {
                        "height": "30px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <i
                      className="fas fa-edit ablue"
                      onClick={[Function]}
                      role="button"
                    />
                  </td>
                  <td
                    style={
                      Object {
                        "height": "30px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <i
                      className="fas fa-trash ablue"
                      onClick={[Function]}
                      role="button"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="row react-bootstrap-table-pagination"
          >
            <div
              className="col-md-6 col-xs-6 col-sm-6 col-lg-6"
            >
              <span
                className="react-bs-table-sizePerPage-dropdown dropdown"
                style={
                  Object {
                    "visibility": "visible",
                  }
                }
              >
                <button
                  aria-expanded={false}
                  className="btn btn-default btn-secondary dropdown-toggle"
                  data-toggle="dropdown"
                  id="scaledtable-pageDropDown"
                  onBlur={[Function]}
                  onClick={[Function]}
                  type="button"
                >
                  10
                   
                </button>
                <ul
                  aria-labelledby="scaledtable-pageDropDown"
                  className="dropdown-menu "
                  role="menu"
                >
                  <a
                    className="dropdown-item"
                    data-page={10}
                    href="#"
                    onMouseDown={[Function]}
                    role="menuitem"
                    tabIndex="-1"
                  >
                    10
                  </a>
                  <a
                    className="dropdown-item"
                    data-page={25}
                    href="#"
                    onMouseDown={[Function]}
                    role="menuitem"
                    tabIndex="-1"
                  >
                    25
                  </a>
                  <a
                    className="dropdown-item"
                    data-page={30}
                    href="#"
                    onMouseDown={[Function]}
                    role="menuitem"
                    tabIndex="-1"
                  >
                    30
                  </a>
                  <a
                    className="dropdown-item"
                    data-page={50}
                    href="#"
                    onMouseDown={[Function]}
                    role="menuitem"
                    tabIndex="-1"
                  >
                    50
                  </a>
                </ul>
              </span>
            </div>
            <div
              className="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
            >
              <ul
                className="pagination react-bootstrap-table-page-btns-ul"
              >
                <li
                  className="active page-item"
                  title="1"
                >
                  <a
                    className="page-link"
                    href="#"
                    onClick={[Function]}
                  >
                    1
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
      <button
        className="mt-4 btn btn-dymium btn-sm"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add mapping
      </button>
    </div>
  </div>,
]
`;
